mapfield.ts

```diff
@@ -34,7 +35,7 @@ export class MapField extends ParseNode {
   keyType: KeyType;
   constructor(
     keyType: KeyType | keyof typeof KeyType,
```

Comment:
Should `keyType` also be a `type`? Perhaps the `Type` class can have a method like `isBaseType` or something?

Reply:
What would the advantage of that be? FWIW, `keyType` is a separate production from `type` in the spec.